dev/buildtool/spin_commands.py

```diff
@@ -133,6 +146,110 @@ def check_file_exists(self, path):
     blob = bucket.get_blob(path)
     return bool(blob)
 
+  def read_file(self, path):
+    """Reads the contents of a GCS file."""
+    bucket = self.__client.get_bucket(self.__bucket)
+    blob = bucket.get_blob(path)
+    return blob.read_as_string()
+
+  def write_file(self, path, contents):
+    """Writes the contents to a GCS file."""
+    bucket = self.__client.get_bucket(self.__bucket)
+    blob = bucket.get_blob(path)
+    return blob.upload_from_string(contents)
+
+  def copy_file(self, source, dest):
+    """Copies the contents to a GCS from from to to."""
+    bucket = self.__client.get_bucket(self.__bucket)
+    blob = bucket.get_blob(source)
+    bucket.copy_blob(blob, bucket, new_name=dest)
+
+
+class PublishSpinCommand(CommandProcessor):
+  """Publish Spin CLI version to the public repository."""
+
+  def __init__(self, factory, options, **kwargs):
+    options_copy = copy.copy(options)
+    options_copy.bom_path = None
+    options_copy.bom_version = None
+    options_copy.git_branch = 'master'
+    options_copy.github_hostname = 'github.com'
+    super(PublishSpinCommand, self).__init__(factory, options_copy, **kwargs)
+
+    check_options_set(options, ['spin_version']) # Ensure we have a version to promote.
+    bom_contents = BomSourceCodeManager.load_bom(options_copy)
+    gate_entry = bom_contents.get('services', {}).get('gate', {})
+    if not gate_entry:
+      raise_and_log_error(
+          ConfigError('No gate service entry found in bom {}'.format(bom_contents)))
+
+    gate_version = gate_entry['version']
+    match = re.match(r'(\d+)\.(\d+)\.(\d+)-\d+', gate_version)
+    if match is None:
+      raise_and_log_error(
+          ConfigError('gate version {version} is not X.Y.Z-<buildnum>'
+                      .format(version=gate_version)))
+    self.__stable_version = '{major}.{minor}.{patch}'.format(
+        major=match.group(1), minor=match.group(2), patch=match.group(3))
+    self.__scm = BranchSourceCodeManager(options_copy, self.get_input_dir())
+    # TODO(jacobkiefer): Add spin CLI autogenerated docs.
+
+    dash = gate_version.find('-')
+    semver_str = gate_version[0:dash]
+    semver_parts = semver_str.split('.')
+    if len(semver_parts) != 3:
+      raise_and_log_error(
+          ConfigError('Expected gate_version in the form X.Y.Z-N'))
+    self.__release_branch = 'release-{maj}.{min}.x'.format(
```

Comment:
The release- version here is the gate version. Everywhere else release- version is the spinnaker version. I think that is going to cause confusion. Either this should be a spinnaker version or use a different string prefix than "release-".

Reply:
Good point.